### PR TITLE
auth API

### PR DESCRIPTION
**Pull request by alrocar**
Closes https://github.com/CartoDB/carto-python/issues/94

I'm doing this as part of a leapfrog. Since this is a public CARTO API we should add proper reference docs and examples. I'll do it in a [different issue](https://github.com/CartoDB/carto-python/issues/126).

### CODE REVIEW

File: carto/api_keys.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    json_collection_attribute = "result"
+    paginator_class = CartoPaginator
+
+    def create(self, name, apis=['sql', 'maps'], tables=None, services=None):
+        """
+        Creates a regular APIKey.
+
+        :param name: The API key name
+        :type name: str
+
```
Comment (oleurud) [inline carto/api_keys.py]: Is not needed to comment on the rest of the parameters?

File: carto/api_keys.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        if not apis:
+            raise CartoException("'apis' cannot be empty. Please specify which CARTO APIs you want to grant. Example: ['sql', 'maps']")
+        grants.append({'type': 'apis', 'apis': apis})
+        if tables and (len(tables) > 0):
+            if isinstance(tables[0], dict):
+                grants.append({'type': 'database', 'tables': tables})
+            elif isinstance(tables[0], TableGrant):
+                grants.append({'type': 'database', 'tables': [x.toJson for x in tables]})
+        if services:
+            grants.append({'type': 'dataservices', 'services': tables})
```
Comment (oleurud) [inline carto/api_keys.py]:
```suggestion
            grants.append({'type': 'dataservices', 'services': services})
```

File: carto/api_keys.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+                        ]
+                    }
+                ]
+            }
+    """
+    schema = CharField()
+    name = CharField()
+    permissions = CharField(many=True)
+
+    def toJson(self):
```
Comment (oleurud) [inline carto/api_keys.py]: Not sure:
```suggestion
    def to_json(self):
```

File: carto/api_keys.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+        """
+        grants = []
+        if not apis:
+            raise CartoException("'apis' cannot be empty. Please specify which CARTO APIs you want to grant. Example: ['sql', 'maps']")
+        grants.append({'type': 'apis', 'apis': apis})
+        if tables and (len(tables) > 0):
+            if isinstance(tables[0], dict):
+                grants.append({'type': 'database', 'tables': tables})
+            elif isinstance(tables[0], TableGrant):
+                grants.append({'type': 'database', 'tables': [x.toJson for x in tables]})
```
Comment (oleurud) [inline carto/api_keys.py]: Not sure
```suggestion
                grants.append({'type': 'database', 'tables': [x.to_json for x in tables]})
```

File: carto/api_keys.py (hunk 2 above)
Comment (alrocar) [reply carto/api_keys.py]: upss 🤦‍♂

File: .travis.yml [hunk 5]
```diff
@@ -1,10 +1,8 @@
 language: python
 python:
 - '2.7'
-- '3.4'
```
Comment (oleurud) [inline .travis.yml]: Thank you

**Pull request merged**